birthday/birthday.py

```diff
@@ -0,0 +1,259 @@
+                    if role_id is not None:
+                        role = discord.utils.get(guild.roles, id=role_id)
+                        if role is not None:
+                            try:
+                                await member.add_roles(role)
+                            except (discord.Forbidden, discord.HTTPException):
+                                pass
+                            else:
+                                async with self.config.guild(guild).yesterdays() as yesterdays:
+                                    yesterdays.append(member.id)
+                    channel = guild.get_channel(guild_config.get("channel"))
+                    if channel is not None:
+                        await channel.send(embed=embed)
+
+    async def clean_bdays(self):
+        birthdays = await self.get_all_date_configs()
+        for guild_id, guild_bdays in birthdays.items():
+            for date, bdays in guild_bdays.items():
+                for user_id, year in bdays.items():
+                    if not any(g.get_member(int(user_id)) is not None for g in self.bot.guilds):
+                        async with self.get_date_config(guild_id, date)() as config_bdays:
+                            del config_bdays[user_id]
+                config_bdays = await self.get_date_config(guild_id, date)()
+                if len(config_bdays) == 0:
+                    await self.get_date_config(guild_id, date).clear()
+
+    async def remove_user_bday(self, guild_id: int, user_id: int):
+        user_id = str(user_id)
+        birthdays = await self.get_guild_date_configs(guild_id)
+        for date, user_ids in birthdays.items():
+            if user_id in user_ids:
+                await self.get_date_config(guild_id, date).get_attr(user_id).clear()
+
+
+    async def clean_yesterday_bdays(self):
+        all_guild_configs = await self.config.all_guilds()
+        for guild_id, guild_config in all_guild_configs.items():
+            for user_id in guild_config.get("yesterdays", []):
+                asyncio.ensure_future(self.clean_bday(guild_id, guild_config, user_id))
+            await self.config.guild(discord.Guild(data={"id": guild_id}, state=None)).yesterdays.clear()
+
+    async def do_today_bdays(self):
+        guild_configs = await self.get_all_date_configs()
+        for guild_id, guild_config in guild_configs.items():
+            this_date = datetime.datetime.utcnow().date().replace(year=1)
+            todays_bday_config = guild_config.get(str(this_date.toordinal()), {})
+            for user_id, year in todays_bday_config.items():
+                asyncio.ensure_future(self.handle_bday(int(user_id), year))
+
+    async def maybe_update_guild(self, guild: discord.Guild):
+        if not guild.unavailable and guild.large:
+            if not guild.chunked or any(m.joined_at is None for m in guild.members):
+                await self.bot.request_offline_members(guild)
+
+    def parse_date(self, date_str: str):
```

Comment:
I would take a look at the parse_time function in the punish/time_utils for parsing data, since im sure you don't have to have that format for the datautil.parser to parse it. The parser for dateutil is very robust, compared to just getting the date using datetime.